v3/src/components/axis/hooks/use-sub-axis.ts

```diff
@@ -95,8 +95,8 @@ export const useSubAxis = ({
         },
         renderNumericAxis = () => {
           const numericScale = multiScale.scaleType === "linear"
-                                ? multiScale.numericScale as ScaleLinear<number, number>
-                                : undefined
+                                 ? multiScale.numericScale?.range(newRange) as ScaleLinear<number, number>
```

Comment:
This seems wrong:
1. It shouldn't be the job of the `renderNumericAxis` function of the `useSubAxis` hook to update the range of the scale owned/maintained by the `MultiScale`.
2. My recollection of the way this was supposed to work originally is that the scale owned/maintained by the `MultiScale` was responsible for mappings to the entire graph, while code responsible for subplots would make _a_ copy of that scale and adjust the range accordingly. We seem to be heading down a path in which there is no distinction between the scale owned/maintained by the `MultiScale` and the copies used by the subplots, but the code for subplots requires a blizzard of copies of that scale anyway.

I think there needs to be a larger consideration of the roles, responsibilities, and use of scales in the graph code.

Reply:
@kswenson Sorry. I thought this would be a quick fix for a bug that's making it difficult to continue the bar chart work. Before the changes made [here](https://github.com/concord-consortium/codap/pull/1166) the range was being updated [here](https://github.com/concord-consortium/codap/blob/b658aa22f65c2f23a7d0625bc1c748dd1beb062a/v3/src/components/axis/hooks/use-sub-axis.ts#L76) in `useSubAxis`, although now I see that was being done to a copy made using `copy()`. Would using a copy and setting the copy's range here go at least some way toward addressing the issues you're concerned with? If not, I'm not sure I should devote more time to this right now.

Reply:
Yes, restricting the scale changes to scales created/maintained by this code would ameliorate the architectural issues raised above. The seemingly excessive creation of copies is a potential performance issue rather than an architectural one.

Reply:
OK, thanks. I've made that change and drafted a [new story](https://www.pivotaltracker.com/story/show/187317584) about further investigation into the use of copied scales.